rwlib/source/fonts/Unicode.hpp

```diff
@@ -0,0 +1,302 @@
+#ifndef _RWLIB_FONTS_UNICODE_HPP_
+#define _RWLIB_FONTS_UNICODE_HPP_
+
+#include <cstdint>
+#include <iosfwd>
+#include <sstream>
+
+/**
+ * unicode_t represent a unicode data point. (UTF-32)
+ */
+using unicode_t = std::uint32_t;
```

Comment:
Can we not use char32_t? these are just UTF-32 encoded characters right?

Reply:
Yup we can and we will.